src/net/mqtt_interface.rs

```diff
@@ -0,0 +1,235 @@
+use crate::hardware::{
+    design_parameters::MQTT_BROKER, CycleCounter, EthernetPhy, NetworkStack,
+};
+
+use core::{cell::RefCell, fmt::Write};
+
+use heapless::{consts, String};
+use serde::Serialize;
+
+use super::{Action, MqttMessage, SettingsResponse, SettingsResponseCode};
+
+/// MQTT settings interface.
+pub struct MqttInterface<S>
+where
+    S: miniconf::Miniconf + Default + Clone,
+{
+    telemetry_topic: String<consts::U128>,
+    default_response_topic: String<consts::U128>,
+    mqtt: RefCell<minimq::MqttClient<minimq::consts::U256, NetworkStack>>,
+    settings: RefCell<S>,
```

Comment:
Why do we need `RefCell` for these? All methods that need the mutable ref to the client have a mutable ref to `self` already.